sourcemash/frontend/static/js/views/register_modal_view.js

```diff
@@ -0,0 +1,46 @@
+Sourcemash.Views.RegisterModalView = Backbone.View.extend({
+    template: JST['register-modal'],
+
+    events: {
+        'submit #register': 'registerFromModal',
+        'click .register-close': 'activateLoginForm'
+    },
+
+    registerFromModal: function(e) {
+        e.preventDefault();
+        var formData = JSON.stringify($("#register").serializeObject());
+        $('#register-password').val('');
+        var posting = $.ajax({
+                          type: "POST",
+                          url: "/register",
+                          data: formData,
+                          success: this.registerUserOrShowErrors,
+                          contentType: "application/json"
+                      });
+    },
+
+    registerUserOrShowErrors: function(data) {
+        var user = data.response.user;
+        if (user) {
+          mixpanel.track("Created Account");
+          $("#register-modal").closeModal();
+          toast("Check your email for confirmation!", 2000);
+          setTimeout(function(){window.location.replace("/")}, 2000);
+        };
+        var errors = data.response.errors;
+        if (errors) {
+          errorMsg = errors.email || errors.password || {};
+          $("#register-errors").html(errorMsg);
+        };
+    },
+
+    activateLoginForm: function() {
+      toast("<-- Login over there!", 3000)
```

Comment:
If you play with it, you’ll see it’s confusing where to go. The toast appears because you clicked “I already have an account”, and it’s weird if the user thinks nothing happened except the modal closing when they clicked that